### ISSUE TITLE
Hero meeting screen, "fast move army" buttons should transfer creatures in respect to their power

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#4394]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Currently, move army buttons move the entire army to a meeting hero starting from the left slot up to the right one (also this feature unites stack of the same type, if there are any):
![image](https://user-images.githubusercontent.com/55348946/136847335-3f5cec9e-fa63-4e6e-b065-2804ae237251.png)
![image](https://user-images.githubusercontent.com/55348946/136847356-d10060a7-6301-4407-89ee-15215a28bcff.png)

### Describe the possible solution

This logic is inconvenient. The most common usage of this button is to give army from one hero to another making the strongest possible army. 
For example: we should transfer Titans first, then Rocs, then Boars and the last one (if we have free space for them) Halflings.
So in previous screenshots the result should be: 
![image](https://user-images.githubusercontent.com/55348946/136846750-ae94aa93-f550-493f-92ff-54268f21d953.png)
![image](https://user-images.githubusercontent.com/55348946/136847293-11a6f915-e27b-43b2-92b9-03dcc793e59a.png)

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog and @ihhub,

I think the default action should be to keep one creature that is the fastest (in priority) AND the weakest.
Generally, this is what I do and what the majority of players I've observed in let's play do, which is normal because it's better to have a fast primary (at least temporarily)/secondary hero.

For the selection of the creature/troop to keep, I made an enhancement proposal in this way here: https://github.com/ihhub/fheroes2/discussions/3742. :)

**Comment by Branikolog** [comment]
@LeHerosInconnu 

> I think it is not possible to cover 100% of the cases.

100% agreed. :)
Me personally consider the situation we're talking about as "Army transferring", so the main goal is to transfer as much army, as we can. (Not making an initial hero as mobile, as possible (which is often a secondary purpose.) ) Which means, that the strongest creatures should be transferred and we weakest should be left in initial army.

> It is therefore complicated to determine the player's goal when he transfers his armies between heroes.

That's true. We cannot read players thoughts, but we should aim to allow player translate this thoughts into realty as much predictable and obvious, as we can.

I think it's impossible to implement such logic, which chooses the most mobile and the less valuable creature to be left within this issue, because it is super-complex and still cannot reveal players goals for sure.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> Me personally consider the situation we're talking about as "Army transferring", so the main goal is to transfer as much army, as we can. (Not making an initial hero as mobile, as possible (which is often a secondary purpose.) ) Which means, that the strongest creatures should be transferred and we weakest should be left in initial army.

Here is what I have noticed (and what I also practice in most cases).
Generally when a player (in the let's plays I've watched) performs this action of transferring his troops from a secondary hero to the main hero (even without using the arrows or other system at his disposal, so moving the troops one by one), he tries to have the most powerful army for his main hero AND AT THE SAME TIME he tries to have the secondary hero with the highest movement speed.
And when a player transfers his troops from the main hero to a secondary hero (no combat in perspective for example), he tries to have the main hero with the highest movement speed.
What we notice here is that the player makes sure that his heroes keep the highest possible movement speed in all cases.
